jerry-core/parser/js/js-scanner-util.c

```diff
@@ -1688,10 +1690,17 @@ scanner_is_context_needed (parser_context_t *context_p, /**< context */
       continue;
     }
 
-    if (JERRY_UNLIKELY (check_type == PARSER_CHECK_FUNCTION_CONTEXT)
-        && (SCANNER_STREAM_TYPE_IS_ARG (type) || SCANNER_STREAM_TYPE_IS_ARG_FUNC (type)))
+    if (JERRY_UNLIKELY (check_type == PARSER_CHECK_FUNCTION_CONTEXT))
     {
-      continue;
+      if (SCANNER_STREAM_TYPE_IS_ARG (type))
+      {
+        continue;
+      }
+
+      if (SCANNER_STREAM_TYPE_IS_ARG_FUNC (type))
+      {
+        return true;
```

Comment:
This case is triggered, if a function has an argument, which is initialized with a function or eval, and there is another argumnt, which has the same name as a function statement in the function scope. While this check could be more sophisticated, I think it is not worth it unless omebody wants it.

Reply:
I'd prefer this comment in the code as well.

Reply:
Good idea. I have been able to make the check a bit better. Now the corner case (returning with true when unnecessary) is extremely rare.